plugins/GSdx/Renderers/HW/GSTextureCache.cpp

```diff
@@ -2077,26 +2050,10 @@ void GSTextureCache::SourceMap::RemoveAt(Source* s)
 	}
 	else
 	{
-		for(size_t i = 0; i < countof(m_pages); i++)
+		s->m_pages.loopPages([this, s](uint32 page)
 		{
-			if(uint32 p = s->m_pages_as_bit[i])
-			{
-				auto* m = &m_map[i << 5];
-				const auto* e = &s->m_erase_it[i << 5];
-
-				unsigned long j;
-
-				while(_BitScanForward(&j, p))
```

Comment:
This was a micro optimization from greg for reducing the source cache invalidation time with the old page validity bitmap; It might be a good benchmark to compare the time spent in the GSTextureCache::SourceMap::RemoveAt before and after the modifications.

Reply:
Hmm the two commits that modified it before were fa1377a8ee5120522a47666fb4416b45fd85c3dd and e58776e0951fd36a8cd0d03eda7df19e19b2386c

fa1377a8ee5120522a47666fb4416b45fd85c3dd looks like it was adding a caching system that's since been replaced, though it does name Champion of Norrath as the game to profile for this section of code

e58776e0951fd36a8cd0d03eda7df19e19b2386c is the one that adds the `_BitScanForward` and doesn't mention what games it's trying to improve

So does anyone have Champion of Norrath (or a gsdump of it) to profile?

Reply:
Champions of Norrath is just a Snowblind engine game, so any game using that, for example Baldurs Gate, will probably do.

Reply:
My memory has betrayed me about the git history, sorry. Anyway I tested the beginning of Baldur's Gate: Dark Alliance, and it is even a tad faster now! Great job.